python/pyarrow/public-api.pxi

```diff
@@ -416,3 +416,42 @@ cdef api object pyarrow_wrap_batch(
     cdef RecordBatch batch = RecordBatch.__new__(RecordBatch)
     batch.init(cbatch)
     return batch
+
+
+cdef api bint pyarrow_is_function_registry(object registry):
```

Comment:
Are these meant to be exposed to third-party C++ or Cython code? Otherwise we shouldn't define such public (un)wrapping functions.